cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicController.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.strimzi.controller.cluster.ClusterController;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the topic controller deployment
+ */
+public class TopicController extends AbstractCluster {
+
+    public static final String KIND = "topic";
+
+    private static final String NAME_SUFFIX = "-topic-controller";
+
+    // Topic Controller configuration
+    protected TopicControllerConfig config;
+
+    // Port configuration
+    protected static final int HEALTHCHECK_PORT = 8080;
+    protected static final String HEALTHCHECK_PORT_NAME = "hcheck-port";
```

Comment:
`hcheck-port` seems a bit strange name ... IMHO `healthcheck` or `healthcheck-port` would be better?

Reply:
I was `healthcheck-port` then it turned out that the limit is 15 characters in Kubernetes ... but `healthcheck` should work :-)